### ISSUE TITLE
[$500] New user cannot see old user's first message after login, until old user sends another message

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#26052]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

## Action Performed:
1. Create a new account
2. Send a message from an old account to the new account
3. View the message from the new account
4. The message doesn't show in the new account 
5. Send another message from the old account
6. The initial message is then visible in the new account

## Expected Result:
The new user should be able to see the first message right away / no inconsistencies when viewing messages 

## Actual Result:
The new user cannot see the old user's first message after login until the old user sends another message

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android / native
- [ ] Android / Chrome
- [ ] iOS / native
- [ ] iOS / Safari
- [x] Windows / Chrome 
- [ ] MacOS / Desktop

**Version Number:** 1.3.57-5

**Reproducible in staging?:** Yes

**Reproducible in production?:** Yes

**If this was caught during regression testing, add the test name, ID and link from TestRail:**

**Email or phone of affected tester (no customers):**

**Logs:** https://stackoverflow.com/c/expensify/questions/4856

**Notes/Photos/Videos:** Any additional supporting documentation

https://github.com/Expensify/App/assets/78819774/f2603889-ec9c-483b-8a72-728caca9a58c

https://github.com/Expensify/App/assets/78819774/42d9d44b-47b6-4922-8c31-7ccea8c0b1cf

**Expensify/Expensify Issue URL:**

**Issue reported by:** @avi-shek-jha

**Slack conversation:** https://expensify.slack.com/archives/C049HHMV9SM/p1692472541452829

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01e715cd77e5235004</li>
        <li>Upwork Job ID: 1699183839647227904</li>
        <li>Last Price Increase: 2023-09-19</li>
    </ul>
</details>

**Comment by maddylewis** [comment]
need to test in Windows / not able to log into BrowserStack using infra@ credentials. coming back to this one.

**Comment by misgana96** [comment]
I reported this bug earlier in Slack [https://expensify.slack.com/archives/C049HHMV9SM/p1692245063170379](here)

**Comment by maddylewis** [comment]
heya @0xmiroslav - i've been unable to reproduce this as i don't have access to Windows and haven't been able to log into the browserstack environment where Windows is accessible. Are you able to repro this on your end? Lmk - ty!

**Comment by maddylewis** [comment]
@0xmiroslav - thanks for confirming. was also able to repro but just not on windows. 

> Another bug to fix here is: new user cannot see old user's first message after login, until old user sends another message.

should the fix for that be included in this issue or should that be separate?

**Comment by maddylewis** [comment]
alright, i updated the title and OP to reflect the bug you described.

**Comment by maddylewis** [comment]
OP is updated / no proposals yet. I will up the bounty on Wednesday if there's still no interest.

**Comment by OlimpiaZurek** [comment]
Hi, I'm Olimpia from Callstack - expert contributor group. I’d like to work on this issue.

**Comment by maddylewis** [comment]
@OlimpiaZurek - bump on proposal! ty!

**Comment by OlimpiaZurek** [comment]
I've been working on another issue for the past few days. I'll start investigating this issue this week.

**Comment by maddylewis** [comment]
we're waiting on @OlimpiaZurek's proposal and then @0xmiroslav will review

**Comment by maddylewis** [comment]
@0xmiroslav - lmk your thoughts. it looks like that linked PR is still in review so im not sure how that would fix this?

**Comment by maddylewis** [comment]
backend fix is in review - https://github.com/Expensify/App/issues/26052#issuecomment-1731501213

**Comment by mvtglobally** [comment]
Issue not reproducible during KI retests. (First week)

**Comment by maddylewis** [comment]
this may be fixed https://github.com/Expensify/App/issues/26052#issuecomment-1738374788

will hold on closing until that's confirmed.

**Comment by maddylewis** [comment]
will close on Oct 5 if not reproducible - https://github.com/Expensify/App/issues/22480#issuecomment-1723788324
